tests/test_regrid/test_regridder.py

```diff
@@ -32,6 +32,174 @@ def quads(dx):
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def grid_b():
+    return xr.DataArray(
+        data=np.zeros(16).reshape((4, 4)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([175, 125, 75, 25]),
+            "x": np.array([25, 75, 125, 175]),
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def expected_results_centroid():
+    return xr.DataArray(
+        data=np.array(
+            [
+                np.nan,
+                np.nan,
+                np.nan,
+                np.nan,
+                np.nan,
+                0,
+                1,
+                np.nan,
+                np.nan,
+                3,
+                4,
+                np.nan,
+                np.nan,
+                np.nan,
+                np.nan,
+                np.nan,
+            ]
+        ).reshape((4, 4)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([175, 125, 75, 25]),
+            "x": np.array([25, 75, 125, 175]),
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def expected_results_overlap():
+    # --------
+    # target | source            | ntarget | sum(source)/ntaget
+    # 0      | 0                 | 1       | 0.0
+    # 1      | 0 1               | 2       | 0.5
+    # 2      |   1 2             | 2       | 1.5
+    # 3      |     2             | 1       | 2.0
+    # 4      | 0     3           | 2       | 1.5
+    # 5      | 0 1   3 4         | 4       | 2.0
+    # 6      |   1 2   4 5       | 4       | 3.0
+    # 7      |     2     5       | 2       | 3.5
+    # 8      |       3     6     | 2       | 4.5
+    # 9      |       3 4   6 7   | 4       | 5.0
+    # 10     |         4 5   7 8 | 4       | 6.0
+    # 11     |           5     8 | 2       | 6.5
+    # 12     |             6     | 1       | 6.0
+    # 13     |             6 7   | 2       | 6.5
+    # 14     |               7 8 | 2       | 7.5
+    # 15     |                 8 | 1       | 8.0
+    # --------
+    return xr.DataArray(
+        data=np.array(
+            [
+                0.0,
+                0.5,
+                1.5,
+                2.0,
+                1.5,
+                2.0,
+                3.0,
+                3.5,
+                4.5,
+                5.0,
+                6.0,
+                6.5,
+                6.0,
+                6.5,
+                7.5,
+                8.0,
+            ]
+        ).reshape((4, 4)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([175, 125, 75, 25]),
+            "x": np.array([25, 75, 125, 175]),
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def expected_results_linear():
+    # --------
+    # target | source            | weights            | sum(source x weight)
+    # 5      | 0 1   3 4         | 0.25 0.25 0.25 0.25    | 2.0
```

Comment:
for completeness, it would be nice to have an example where the source data also contains some NaN value.